task: update readme to include dev workflow (#520)
b2io/base2.io: README.md

   <details>
    <summary>If a new card is needed</summary>   
    If you need to create a new card then click the `Create` button in Jira's header and select the correct card type

     - **story:** A user-focused feature written as a user story.
     - **task:** A development task or chore that isn't written as a user story.
     - **fix:** A task that corrects an issue with the application.

   </details>

2. Assign yourself to the card
3. Track time in Harvest via Jira by clicking the `Open Harvest Time Tracking` option next to `Harvest Time Tracking` in the Jira card. This will start a timer in Harvest with the card's name and a link to the card.
4. Update the card's status to be `In Progress`
5. Create a new branch off of the `next` branch prefixed with the card id (e.g. `B2-99-short-desc-of-task`). Note that the capitalization of the ID is important. This will cause Jira to recognize the card and begin tracking it within a `Development` section of the card.
6. When committing add `#comment` within the contents of the commit message if you'd like it to be included in the Jira card as a commit.
7. When creating a PR

>    1. Ensure that it's being PRed into the correct base branch (e.g. `next`)

Reviewer: Small potatoes, just a possible re-wording. Feel free to play with this, I just think the word "destination" could be helpful here.

```suggestion
   1. Ensure that the PR's destination is the correct base branch (e.g. `next`)
```

>    2. Give it a title prefixed with it's card type and a short description (e.g. `task: update readme to include dev workflow`)

Reviewer: ```suggestion
   2. Give it a title prefixed with its card type and a short description (e.g. `task: update readme to include dev workflow`)
```

Author: 🤕

   3. Provide a description that explains what was changed, why it was changed, and any other information that may help the reviewer (e.g. screenshots or proof of successful tests)

>    4. Include the Jira card number, a link to the card number, and `#comment` to communicate which card it closes and to link the commit in Jira. E.g.:

Reviewer: Here's an example of the `#comment` syntax from my other comment. Maybe we could move this example up to the first section, and hyperlink to it here?

Reviewer: Also, the comment here ("Update readme to include...") seems like the sort of thing that would be in the description of the PR. Do we want it in both places?

This may be a conversation for another time/place, I just haven't seen the `#comment` pattern before, so I'm curious how we'd like to use it and such.

Author: This example is the PR description example. I too am trying out the `#comment` pattern. I think that I'm going to merge as-is (with the other suggestions accounted for) and then see where things land with the `#comment` approach after a bit more use. I fully expect this readme to be touched a few more times.
